x-pack/plugins/observability/public/pages/rules/index.tsx

```diff
@@ -117,20 +117,19 @@ function RulesPage() {
       pageHeader={{
         pageTitle: <>{RULES_PAGE_TITLE}</>,
         rightSideItems: [
-          authorizedToCreateAnyRules && (
-            <EuiButton
-              iconType="plusInCircle"
-              key="create-alert"
-              data-test-subj="createRuleButton"
-              fill
-              onClick={() => setCreateRuleFlyoutVisibility(true)}
-            >
-              <FormattedMessage
-                id="xpack.observability.rules.addRuleButtonLabel"
-                defaultMessage="Create rule"
-              />
-            </EuiButton>
-          ),
+          <EuiButton
```

Comment:
This reduces the 'pop in' effect on this page where buttons suddenly jump to the left

Reply:
Well @maciejforcone, what do you think?

Reply:
I checked how it works in Rules. So we do show "manage rules" always, but when user gets there and has no permissions to Create rule - then we don't show the button. But for example in Monitor view we show the button, but later in form we show the message "permissions needed". 
<img width="203" alt="image" src="https://user-images.githubusercontent.com/104486848/203773394-eed98c70-0f66-4366-be1c-7803654b7e79.png">
<img width="893" alt="image" src="https://user-images.githubusercontent.com/104486848/203773610-15a40997-370a-4621-a99c-90249cfb82a4.png">


For some actions we also show disabled button 
<img width="358" alt="image" src="https://user-images.githubusercontent.com/104486848/203773507-47805269-4dc5-48e7-bd0a-bc87881a956e.png">

So looks like we don't have a common pattern for that yet. I'd stick with what we have for now - showing the button, and then in for we show the info of missing permissions. And in case when there is instant action like this toggle:
<img width="191" alt="image" src="https://user-images.githubusercontent.com/104486848/203774040-7dd630f4-a0e4-4f16-b834-2cbb27208aeb.png">
the button can be disabled, with missing permissions info on hover.

Reply:
Ok, @maciejforcone agrees that a less jumpy UI is better. @maryam-saeidi hope this answers your question.

Reply:
And to the question of loader vs element - less jumps is always better, so I like how @CoenWarmer solved that.

Reply:
@maciejforcone Thanks for the input, the loader option was for the whole page/title (not the button). I think I saw a similar approach on the cases page if I am not mistaken, so it will also not be jumpy. But if we use disable approach, that's also a good option.